### PR TITLE
Fix file handlers improperly matching some file types

### PR DESCRIPTION
**Pull request by djhoese**
I noticed this while working on other issues. The file handler's `available_datasets` method uses a method to check if the current file handler's file type matches the file type for the current dataset being evaluated. The issue is that the way the condition was written it was producing a false positive where it was matching file types with similar names. For example, `if "file_type_name" in "file_type_name2"`. This is only suppose to match if it matches exactly, but the if statement was assuming the second parameter was a list/container.

This PR fixes this issue and "adds" tests by putting one of the fake readers used in multiple tests so that it matches the failing real world case I ran in to.

 - [x] Tests added

### CODE REVIEW

**Review by pnuu — COMMENTED** [review]
> This PR fixes this issue and "adds" tests by putting one of the fake readers used in multiple tests so that it matches the failing real world case I ran in to.

I'm not sure I understand what you are saying here.

I also don't exactly understand how this bit works, but shouldn't there be a new test for testing that passing a sub-string of a valid `ds_ftype` returns `None` if it doesn't exactly match anything? I don't see this would be tested by just adjusting a file type for a dataset and not modifying any of the tests themselves.

File: satpy/readers/file_handlers.py [hunk 1]
```diff
@@ -157,7 +157,7 @@ def file_type_matches(self, ds_ftype):
         """
         if isinstance(ds_ftype, str) and ds_ftype == self.filetype_info['file_type']:
             return True
-        elif self.filetype_info['file_type'] in ds_ftype:
+        if isinstance(ds_ftype, (list, tuple)) and self.filetype_info['file_type'] in ds_ftype:
```
Comment (mraspaud) [inline satpy/readers/file_handlers.py:160]: Would it be clearer/simpler with this?
```python
        if isinstance(ds_ftype, str):
            ds_ftype = [ds_ftype]
        if self.filetype_info['file_type'] in ds_ftype:
```

**Review by mraspaud — APPROVED** [review]
Just a small styling comment, otherwise lgtm!

File: satpy/readers/file_handlers.py (hunk 1 above)
Comment (mraspaud) [reply satpy/readers/file_handlers.py:160]: Or even
```python
        if isinstance(ds_ftype, str):
            ds_ftype = [ds_ftype]
        return self.filetype_info['file_type'] in ds_ftype
```

**Comment by djhoese** [comment]
@pnuu You aren't wrong. I should add a test specifically for this, but I was trying to cheat. There are fake reader YAML files in the tests directory that are used by multiple tests to make sure that things are loaded correctly (reader datasets, available_dataset_names, compositors, modifiers, loading modified compositors, etc). What I've done here is similar to adding an additional check to an existing test. I modified the "input" of the test just slightly so it didn't change the behavior of the test, but did change if the current bug would be hit. Like noticing that your test doesn't have a `standard_name` defined on the input, but having it defined adds an additional path in the code that is taken when executed.

Anyway...I'll add a test.

File: satpy/readers/file_handlers.py (hunk 1 above)
Comment (djhoese) [reply satpy/readers/file_handlers.py]: Done. Along with the tests @pnuu asked for.

**Pull request merged**